dotcom-rendering/src/web/components/SecureSignup.tsx

```diff
@@ -0,0 +1,163 @@
+import createCache from '@emotion/cache';
+import { CacheProvider, css } from '@emotion/react';
+import createEmotionServer from '@emotion/server/create-instance';
+import { neutral, space, text, textSans } from '@guardian/source-foundations';
+import {
+	Button,
+	Label,
+	Link,
+	TextInput,
+} from '@guardian/source-react-components';
+import { renderToString } from 'react-dom/server';
+import { Island } from './Island';
+import { SecureSignupIframe } from './SecureSignupIframe.importable';
+
+// The Google documentation specifies that if the 'recaptcha-badge' is hidden,
+// their T+C's must be displayed instead. <SecureSignupIframe> hides the
+// badge, so <RecaptchaTerms> must be displayed with it.
+// https://developers.google.com/recaptcha/docs/faq#id-like-to-hide-the-recaptcha-badge.-what-is-allowed
+
+type Props = { newsletterId: string; successText: string };
+
+const termsStyle = css`
+	${textSans.xxsmall()}
+	color: ${text.supporting};
+	a {
+		${textSans.xxsmall()}
+		text-decoration: none;
+		:hover {
+			text-decoration: underline;
+		}
+	}
+	strong {
+		color: ${neutral[0]};
+		font-weight: bold;
+	}
+`;
+
+const PrivacyTerms = () => {
+	return (
+		<span css={termsStyle}>
+			<strong>Privacy Notice: </strong>
+			Newsletters may contain info about charities, online ads, and
+			content funded by outside parties. For more information see our{' '}
+			<Link
+				href="https://www.theguardian.com/help/privacy-policy"
+				rel="noopener noreferrer"
+			>
+				privacy policy
+			</Link>
+			.&nbsp;
+		</span>
+	);
+};
+
+const RecaptchaTerms = () => (
+	<span css={termsStyle}>
+		We operate Google reCaptcha to protect our website and the Google{' '}
+		<Link
+			href="https://policies.google.com/privacy"
+			rel="noopener noreferrer"
+		>
+			Privacy Policy
+		</Link>{' '}
+		and{' '}
+		<Link
+			href="https://policies.google.com/terms"
+			rel="noopener noreferrer"
+		>
+			Terms of Service
+		</Link>{' '}
+		apply.
+	</span>
+);
+
+const generateForm = (
+	newsletterId: string,
+): { html: string; styles: string } => {
+	const cache = createCache({ key: 'email-signup-iframe' });
+
+	// eslint-disable-next-line @typescript-eslint/unbound-method -- not a react method, 'this' binding not required?
+	const { extractCriticalToChunks, constructStyleTagsFromChunks } =
+		createEmotionServer(cache);
+
+	const html = renderToString(
```

Comment:
It'd be great to have a few comments here briefly explaining what's happening here and why, perhaps with a link to this PR to provide more information if needed!

Reply:
Will do - it's potentially a function that could be generalised for use elsewhere, so good to outline what it does.